lib/health_check/utils.rb

```diff
@@ -31,7 +31,14 @@ def self.process_checks(checks, called_from_middleware = false)
           when "emailconf"
             errors << HealthCheck::Utils.check_email if HealthCheck::Utils.mailer_configured?
           when "migrations", "migration"
-            if defined?(ActiveRecord::Migration) and ActiveRecord::Migration.respond_to?(:check_pending!)
+            if defined?(ActiveRecord::Migration) and ActiveRecord::Migration.respond_to?(:check_all_pending!)
+              # Rails 4+
```

Comment:
This PR would have saved us some pain, nice one! Just a minor suggestion:
```suggestion
              # Rails 7.2+
```